manual/source/classes/contributed_definitions/ellipsometry-structure.rst

```diff
@@ -0,0 +1,179 @@
+.. _Ellipsometry-Structure:
+
+========================
+Optical Spectroscopy
+========================
+
+.. index::
+   Ellipsometry
+   DispersiveMaterial
+
+
+.. _Ellipsometry:
+
+Ellipsometry
+##############
+
+Ellipsometry is an optical characterization method to describe optical properties of interfaces and thickness of films.
+The measurements are based on determining how the polarization state of light changes upon transmission and reflection.
+Interpretation is based on Fresnel equations and numerical models of the optical properties of the materials.
+
+In the application definition we provide a minimum set of description elements allowing for a reproducible recording of ellipsometry measurements. 
+
+
+Application Definitions
+-----------------------
+
+    :ref:`NXopt`:
+       A generic application definition for optial spectorscopy measurements, including complex systems up to variable angle spectroscopic ellipsometry. 
+
+    :ref:`NXellipsometry`:
+       An application definition for ellipsometry measurements, including complex systems up to variable angle spectroscopic ellipsometry. 
+
+
+Base Classes
+------------
+
+There is a set of base classes for describing an optical experiment.
+
+    :ref:`NXbeam_path`
+       A beam path consisting of one or more optical elements.
+       
+       NXbeam_path is used in NXopt to describe the beam path, i.e. the arrangement
+       of optical elements between the excitation source and the sample, or between
+       the sample and the detector unit.
+              
+    :ref:`NXbeam_splitter`
+       A beam splitter, i.e. a device splitting the light into two or more beams.
+       
+       Use two or more NXbeam_paths to describe the beam paths after the beam
+       splitter. In the dependency chain of the new beam paths, the first elements
+       each point to this beam splitter, as this is the previous element.
+
+    :ref:`NXfiber`
+       An optical fiber, e.g. glass fiber.
+
+    :ref:`NXlens_opt`
+       Description of an optical lens.
+       
+    :ref:`NXpolarizer_opt`
+       A spin polarizer.
+
+    :ref:`NXwaveplate`
+       A waveplate or retarder.
+
+    :ref:`NXenvironment`
+       Specify external parameters that have influenced the sample,
+       such as the surrounding medium, and varied parameters e.g.
+       temperature, pressure, pH value, optical excitation etc.
+
+
+
+.. _DispersiveMaterial:
+
+Dispersive Material
+###################
+
+A dispersive material is a description for the optical dispersion of materials.
+This description may be used to store optical model data from an ellipsometric analysis 
+(or any other technique) or to build a database of optical constants for optical properties of materials.
+
+Application Definition
+----------------------
+
+    :ref:`NXdispersive_material`:
+       An application definition to describe the dispersive properties of a material.
```

Comment:
Indeed, I designed this to allow exporting optical models from ellipsometry measurements. There the data collection and analysis is often combined with the instrument softwares so it made sense to me to write it out in a standardised way. I do think it is extremely important as the optical models are the actual currency of ellipsometry data and easily being able to reuse them would be really helpful. It is, however, just a part of the whole picture as one would also need a description of sample layers and roughnesses to completely describe such analysis.

I also thought that this eventually could go to /entry/process to basically attach it to a measurement. But I wasn't really sure how to build such a structure in nexus. But I did want to contribute it here already, because it allows you to collect optical properties of a compound. I used this for example to parse the https://refractiveindex.info database which allows me to re-publish this data in a structured and machine readable way (in nomad for example).

Reply:
If you check out how processed results are recorded in, eg., `NXtomoproc` or `NXxasproc`, you see that there is an `NXprocess` group to provide some provenance of the results. I think you could add this too.

Reply:
Yes, this makes sense. I see there is also `NXparameters` to collect parameter information. It might also be useful to see whether this might be used to store the formula parameters/variables.

Generally, the dispersive material is just a precursor for a more expressive construct, i.e., it would be nice to store information of a whole material stack with multiple dispersions. This is typically collected from ellipsometry programs and be store in `NXprocess` with appropriate provenance. But I agree that it makes sense to allow collecting this information at this stage, too.

Reply:
My suggestion is: I will add the additional provenance fields but I think using `NXparameters` needs more work since I split this in single and repeated parameters for dispersive material. Therefore, it involves more time to design this properly.

Reply:
I added the changes as I suggested @PeterC-DLS. I think it should be helpful to collect this information but certainly needs to be updated by us in the future.